src/framework/audio/devtools/waveformmodel.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef MU_AUDIO_WAVEFORMMODEL_H
+#define MU_AUDIO_WAVEFORMMODEL_H
+
+#include <QObject>
+
+#include "modularity/ioc.h"
+#include "async/asyncable.h"
+
+#include "iplayback.h"
+
+namespace mu::audio {
+class WaveFormModel : public QObject, public async::Asyncable
+{
+    Q_OBJECT
+
+    INJECT(audio, IPlayback, playback)
```

Comment:
It may be worth adding to `IAudioIO` and IoC as well.
Otherwise, it is not clear what this class really depends on and what it can affect (for example, can it control play or add new tracks ...)

Reply:
Not sure, I think that would overlap with the idea of a single entry point. It might be confusing which one to use

Reply:
Instead, we might do that trick

![image](https://user-images.githubusercontent.com/57620349/124270092-04347700-db3c-11eb-8d8e-249f90ef5632.png)

so that you'd have to include "IPlayer" interface directly in the consumer class, in order to use it

Reply:
and obviously you'd need to include only what you actually need